### ISSUE TITLE
Update CSL dependencies and test some reported bugs

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#5629]
**Describe the bug**
A community member running OJS 3.1.2 has reported a few formatting bugs with the Citation Style Languages:

- More than one first name written in full results in no space after the last name:: https://doi.org/10.7557/13.5005 (default style in this journal is Chicago),
- When first names are shortened to initials there is also no space after the last name. In addition it messes up where the initials of the co-authors show up:
  - Several first name initials: https://doi.org/10.7557/3.4689 (default style in this journal is APA)
  - One first name initial: https://doi.org/10.7557/3.2612 (default style in this journal is APA).
- Turabian and Harvard add a 0 to Volume number:
- In https://doi.org/10.7557/3.4643 the Volume number is changed from 11 to 110 (see Turabian, we turned off Harvard for this journal)
- In https://doi.org/10.7557/13.4994 just 0 shows up in the Harvard style (the journal does not use Volume numbers).
- Vancouver and Turabian do not use DOI in the link: JandaLaura A., Radovan Bast, and AspaasPer Pippin. “SMARTool”. Open Science Talk0, no. 33 (March 13, 2020). Accessed March 16, 2020. https://septentrio.uit.no/index.php/OSTalk/article/view/5408.

It's been a while since we updated the [CSL styles](https://github.com/citation-style-language/styles) and it looks like there's been a new release of [citeproc-php](https://github.com/seboettg/citeproc-php/). We should update what we're using then test against the cases described above.

**To Reproduce**
You will need to build out test data resembling each case above.

**What application are you using?**
OJS 3.1.2

**Comment by NateWr** [comment]
Some of the space issues seem to be related to PHP 7.3/7.4. See https://github.com/seboettg/citeproc-php/issues/80.

**Comment by NateWr** [comment]
@asmecher several of these spacing issues were fixed with an update to citeproc-php. I've pushed that change to `master`. What's the best way to make sure this goes out with OJS 3.2.0-2? Do I make a submodule commit in OJS pointing to the `master` branch of citationStyleLanguage?

**Comment by asmecher** [comment]
@NateWr, yes, the submodule update of `plugins/generic/citationStyleLanguage` needs to be updated. I have not (yet) created a `stable-3_2_0` branch within that plugin, because there's been no divergence yet -- but when there is, that branch should be created. The submodule update needs to be committed to the `ojs` repo in both `master` and `stable-3_2_0` branches.

**Comment by AhemNason** [comment]
Can I add one to this? I've noticed that if you're running a bilingual journal, the citation generated appears to run some capitalization rules independent of whatever French-language metadata is entered. I have a journal with the French title: 

> Revue canadienne de linguistique appliquée

But, I'm generating this reference in French: 

> Wouters, I., & Fortier, V. (2020). Tu vas à la cerise les dimanches ? L’importance de travailler la perception et la production de phonèmes en allemand langue étrangère. Revue Canadienne De Linguistique appliquée, 23(1), 23-51. https://doi.org/10.37213/cjal.2020.28661

I checked to see if it was style-specific but it doesn't seem to be. This is a little weird mostly because I can't figure out why it didn't also capitalize "appliquée". Maybe I'm wrong to assume it pulls this metadata from the journal-title field?

**Comment by NateWr** [comment]
What citation style are you using?

**Comment by AhemNason** [comment]
this is APA

**Comment by bibliothekswelt** [comment]
Trailing spaces in author’s name result in a “ghost” abbreviation point in citation formats https://forum.pkp.sfu.ca/t/trailing-spaces-in-authors-name-result-in-a-ghost-abbreviation-point-in-citation-formats/65782

**Comment by NateWr** [comment]
Thanks @bibliothekswelt. I think that's something we should fix at the source, so that whitespace is trimmed in author names before it is saved. I filed an issue at: https://github.com/pkp/pkp-lib/issues/6778

**Comment by kaitlinnewson** [comment]
Hi @bozana, the PRs above are ready for your review.

I had hoped that this would improve support for article numbers in our citation styles, but since `citeproc-php` supports CSL 1.0.1 (not the newer 1.0.2), then I think we can only [use the CSL files from that version branch](https://github.com/citation-style-language/styles/tree/v1.0.1) to ensure they are compatible with the right version. Alternatively, we could try to get the newer CSL files on a case-by-case basis and test them carefully (I found some would break the plugin is we use the wrong version e.g. for chicago style, but not all would).

I've also updated the naming of a couple of the CSL files to match what they now use in the CSL styles repository, and I reordered the styles so they are listed alphabetically in the dropdown in most cases.

**Comment by bozana** [comment]
Hi @kaitlinnewson, I would have a few comments/questions:
- We would then also need a migration script for the renamed styles acs-nano → american-chemical-society and ama → american-medical-association in `plugin_settings`                                                                                                                               
- README and composer.json are not consistent: README links to pkp/citeproc-php but composer.json requires seboettg/citeproc-php. Is PKP ready to be referenced as the official dependency?
- README says "compatible with OJS 3.1.x": this seems outdated, so maybe to fix it while the README is being touched anyway? 

Thanks a lot!

**Comment by kaitlinnewson** [comment]
Hi @bozana, ready for another look.

Since we will eventually have to do a larger migration of file names when moving to CSL 1.0.2, I decided to remove the renaming for now and we can do it all at once later.

I've also updated the README further. I think there is a discrepancy for now with the links because [packagist is referencing the old repo location](https://packagist.org/packages/seboettg/citeproc-php), but the repo now lives in the PKP organization now so we can link to it directly (otherwise it would just redirect anyways).

**Comment by bozana** [comment]
Hi @kaitlinnewson, all merged. Thanks!
Would you maybe ask Alec, what is the plan for the packagist?

**Comment by kaitlinnewson** [comment]
Hi @bozana, I dug in a little more and I don't think we necessarily have to rename the package in packagist, and are actually fine to leave it as-is (and it sounds more complicated to rename it: https://github.com/composer/packagist/issues/47). But I can confirm with Alec (it looks like we may at minimum need to update the contact information there).

**Comment by kaitlinnewson** [comment]
Closing as completed.

## QA Evidence

(Evidence of some of the issues which were not marked as completed previously)

**Turabian citation style, showing no "0" added to the volume in the fixed 3.5**

<img width="856" height="614" alt="Image" src="https://github.com/user-attachments/assets/e4da2469-c17b-44a2-a9c9-412b09794f98" />

---

**Harvard citation style handling a missing volume**

It instead now shows it as a "Preprint", which is part of the CSL style file.
([see this forum post](https://forum.pkp.sfu.ca/t/citation-style-language-plugin-harvard-format-error-in-both-ojs-and-omp/98198/4) - workaround includes adding page numbers)

<img width="856" height="618" alt="Image" src="https://github.com/user-attachments/assets/28b5ea58-1f56-4a14-8006-ca9a49a1c0dd" />

---

**Handling of french journal titles in APA**

In 3.4, showing journal title being capitalized in french (slightly different than the original problem but still incorrect):

<img width="855" height="631" alt="Image" src="https://github.com/user-attachments/assets/a3f5ab4e-d787-4295-b589-29505d1ad5ed" />

In the fixed 3.5, following the entry of the journal title in OJS with only the first word capitalized (correct):

<img width="850" height="597" alt="Image" src="https://github.com/user-attachments/assets/34804b5a-824d-4001-b48f-bd8bdf673286" />

---

**3 or more authors in Harvard citation**

In 3.4, incorrectly showing only the first author:

<img width="851" height="876" alt="Image" src="https://github.com/user-attachments/assets/09bf9f2e-6406-4cb0-a443-4b5599be5646" />



Fixed 3.5, showing "et. al" as expected:


<img width="853" height="713" alt="Image" src="https://github.com/user-attachments/assets/4465e270-3212-49ae-878e-e56a31f2c7d5" />
